src/js/letters/components/DownloadLetterLink.jsx

```diff
@@ -1,132 +1,84 @@
+    // TODO: possibly want to explicitly
+    // check for blob URL support with something like
+    // const blobSupported = !!(/^blob:/.exec(downloadUrl));
     const ie10 = !!window.navigator.msSaveOrOpenBlob;
+    const save = document.createElement('a');
     let downloadWindow;
-    if (!ie10) {
+    const downloadSupported = typeof save.download !== 'undefined';
+    if (!downloadSupported) {
+      // Instead of giving the file a readable name and downloading
+      // it directly, open it in a new window with an ugly hash URL
       downloadWindow = window.open();
     }
+    let downloadUrl;
 
-    // TODO: in addition to the new blank browser window, this tries to
-    // pop up a duplicate window which may or may not be suppressed
-    // by the user's browser content settings.
     apiRequest(
       requestUrl,
-      { method: 'POST' },
+      settings,
       response => {
         response.blob().then(blob => {
           if (ie10) {
             window.navigator.msSaveOrOpenBlob(blob, this.props.letterName);
-            return Promise.resolve();
-          }
-          window.URL = window.URL || window.webkitURL;
-          const downloadUrl = window.URL.createObjectURL(blob);
-          // Make sure blob URLs are supported
-          const blobSupported = !!(/^blob:/.exec(downloadUrl));
-          if (blobSupported) {
-            // Try to give the file a nice name instead of an ugly hash
-            // by creating a new link element and setting its download attribute.
-            const link = document.createElement('a');
-            if (typeof link.download !== 'undefined') {
-              document.body.appendChild(link);
-              // downloadWindow.document.body.appendChild(link);
-              link.style = 'display: none';
-              link.target = '_blank';
-              link.href = downloadUrl;
-              link.download = this.props.letterName;
-              link.click();
+          } else {
+            window.URL = window.URL || window.webkitURL;
+            downloadUrl = window.URL.createObjectURL(blob);
+            if (downloadSupported) {
+              // Give the file a readable name if the download attribute
+              // is supported.
+              save.download = this.props.letterName;
+              save.href = downloadUrl;
+              save.target = '_blank';
+              document.body.appendChild(save);
+              save.click();
+              document.body.removeChild(save);
             } else {
-              // The download attribute is not supported on IE11 or
-              // iOS Safari, so live with the ugly hash name.
               downloadWindow.location.href = downloadUrl;
             }
-            // document.body.removeChild(link);
-            // urlObj.revokeObjectURL(downloadUrl);
-            return Promise.resolve();
           }
-          // Make sure this gets to sentry
-          return Promise.reject(new Error('Cannot download pdf blob'));
```

Comment:
Because I need to figure out exactly what these should be doing instead of guessing :)